docs/awsprivatelink.md

```diff
@@ -223,5 +223,220 @@ expectations of required permissions for these credentials.
+    },
+    {
+      "OutputKey": "PublicSubnetIds",
+      "OutputValue": "subnet-00d5d89ce7b049296,subnet-0253d95c4bcfe7e17,subnet-03a4cfddbf7d9f73a",
+      "Description": "Subnet IDs of the public subnets."
+    },
+    {
+      "OutputKey": "VpcId",
+      "OutputValue": "vpc-0e0b3939a23a5e33e",
+      "Description": "ID of the new VPC."
+    }
+  ]
+]
+```
+
+3. Create a Private link VPC using cloudformation template.
+
+```
+aws --region us-east-1 cloudformation create-stack \
+  --stack-name private-link-vpc \
+  --template-body "file://hack/awsprivatelink/vpc.cf.yaml" \
+  --parameters ParameterKey=AvailabilityZoneCount,ParameterValue=3 ParameterKey=VpcCidr,ParameterValue=10.1.0.0/16
+```
+
+4. Once the cloudformation completes, extract the VPC and subnets from the output.
+
+```
+aws --region us-east-1 cloudformation describe-stacks \
+  --stack-name private-link-vpc --query 'Stacks[].Outputs'\
+[
+  [
+    {
+      "OutputKey": "PrivateSubnetIds",
+      "OutputValue": "subnet-0bf2b8faf1c7d200b,subnet-09d53f09b54c38029,subnet-015d46e5f7fb7055d",
+      "Description": "Subnet IDs of the private subnets."
+    },
+    {
+      "OutputKey": "PublicSubnetIds",
+      "OutputValue": "subnet-00d5d89ce7b049296,subnet-0253d95c4bcfe7e17,subnet-03a4cfddbf7d9f73a",
+      "Description": "Subnet IDs of the public subnets."
+    },
+    {
+      "OutputKey": "VpcId",
+      "OutputValue": "vpc-0e0b3939a23a5e33e",
+      "Description": "ID of the new VPC."
+    }
+  ]
+]
+```
+
+### Create OCP cluster in preexisting Hive VPC
+
+Follow the steps outlines in [OpenShift documentation][openshift-install-aws-existing-vpc].
+
+```yaml
+# install-config.yaml
+...
+platform:
+  aws:
+    region: us-east-1
+    subnets:
+    - # include all the subnets public and private created by cloudformation stack
+    defaultMachinePlatform:
+      zones:
+      - us-east-1a
+      - us-east-1b
+      - us-east-1c
+...
+```
+
+### Setup network peering connection between Hive and Private link VPCs
+
+Create and accept the VPC peering connection using [AWS documentation][aws-vpc-peering]
+
+```
+aws --region us-east-1 ec2 create-vpc-peering-connection \
+  --vpc-id $HIVE_VPC_ID \
+  --peer-vpc-id $PRIVATE_LINK_VPC_ID \
+  --tag-specification "ResourceType=vpc-peering-connection,Tags=[{Key=hive-private-link,Value=owned}]"
+```
+
+```
+aws --region us-east-1 ec2 accept-vpc-peering-connection --vpc-peering-connection-id $VPC_PEERING_CONNECTION_ID
+```
+
+Now update the Route tables using [AWS documentation][aws-vpc-peering-route-tables]
+
+The cloudformation stack creates one route table for all the public subnets, and one route table
+for each private subnet.
+
+First update the private route tables for Hive VPC. Update the routes so that `10.1.0.0/16`
+prefix routes to the VPC peering connection.
+
+Second update the private route tables for Private link VPC. Update the routes so that `10.0.0.0/16`
+prefix routes to the VPC peering connection.
```

Comment:
These CIDRs appear backwards from how they were set up above, where the [hive VPC](https://github.com/openshift/hive/pull/1476/files#diff-5fed0fb5d73ed353edbb9348b0aaefcb3ff6930752f3488acc253d8e70f198d4R276) was `10.0.0.0/16` and the [private link VPC](https://github.com/openshift/hive/pull/1476/files#diff-5fed0fb5d73ed353edbb9348b0aaefcb3ff6930752f3488acc253d8e70f198d4R311) was `10.1.0.0/16`.

(It's entirely possible I'm misunderstanding, and this is intentionally criss-crossed.)

Reply:
we need these VPCs to send traffic to other's range using peering connection, that is why they are adding route for the others network range to themselves.
so we need the Hive VPC route tables to include route for 10.1.0.0/16 private link VPC using the peering connection.
and private link VPC route tables to include route for 10.0.0.0/16 Hive VPC using peering connection.